app/models/relay_state.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'sentry_logging'
+
+class RelayState
+  extend ActiveModel::Naming
+  include ActiveModel::Validations
+  include SentryLogging
+
+  attr_accessor :relay_enum, :url
+
+  RELAY_KEYS = Settings.saml.relays.keys.map(&:to_s).freeze
+  LOGIN_URLS = (Settings.saml.relays.to_h.values + [Settings.review_instance_slug]).compact.freeze
+  LOGOUT_URLS = (Settings.saml.logout_relays.to_h.values + [Settings.saml.logout_relay]).compact.freeze
+  ALL_RELAY_URLS = (LOGIN_URLS + LOGOUT_URLS).freeze
+
+  validates :relay_enum, allow_blank: true, inclusion: { in: RELAY_KEYS, message: '[%<value>s] not a valid relay enum' }
+  validates :url, allow_blank: true, inclusion: { in: ALL_RELAY_URLS, message: '[%<value>s] not a valid relay url' }
+  # TODO: Custom validator that BOTH are not blank
+
+  def initialize(relay_enum: nil, url: nil)
+    @relay_enum = relay_enum
+    @url = url
+
+    unless valid?
+      log_message_to_sentry(
+        'Invalid SAML RelayState!', :error,
+        error_messages: errors.messages, url_whitelist: ALL_RELAY_URLS, enum_whitelist: RELAY_KEYS
+      )
+    end
```

Comment:
just log a message if the model is invalid? is this the intended behavior?

Reply:
As written, yes.  The current design defaults to `default_login_url` & default_logout_url` in the events that `valid? == false`.

I'm open to other opinions though, such as raising and returning some kind of `400` or `422` error code in the event of a bad `success_relay` or `RelayState`.  Basically, failing early V.S. trying to gracefully recover.